python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+
+        print('------------')
+
+        return
+
+    def compare_agn_inputs(self, spr_agn_df, spr_agn_lens_df):
+
+        """
+        This test compares the things that get swapped in directly from the input catalog.
+        """
+        
+        db = om10.DB(catalog=os.path.join(os.environ['TWINKLES_DIR'], 'data', 
+                                          'twinkles_lenses_v2.fits'), vb=False)
+
+        errors_present = False
+        phi_error = False
+        z_lens_error = False
+        z_src_error = False
+        lens_major_axis_error = False
+        lens_minor_axis_error = False
+        lens_sed_error = False
+        errors_string = "Errors in: "
+
+        for lens_gal_row in spr_agn_lens_df.iterrows():
+
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_agn_df.query('lens_galaxy_uID == %i' % u_id)
+            use_lens = db.lenses['LENSID'][np.where(db.lenses['twinklesId'] == 
+                                                    spr_sys_df['twinkles_system'].iloc[0])[0]]
+            lens = db.get_lens(use_lens)
+
+            # These should just be different by a minus sign
+            if lens['PHIE'] + lens_gal_df['positionAngle'] > 0.005:
+                if phi_error is False:
+                    errors_string = errors_string + "\nPosition Angles. First error found in lens_gal_id: %i" % u_id
+                    errors_present = True
+                    phi_error = True
+
+            # Redshifts should be identical
+            if lens['ZLENS'] - lens_gal_df['redshift'] > 0.005:
+                if z_error is False:
+                    errors_string = errors_string + "\nLens Redshifts. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    z_lens_error = True
+
+            if np.max(lens['ZSRC'] - spr_sys_df['redshift'].values) > 0.005:
+                if z_src_error is False:
+                    errors_string = errors_string + "\nSource Redshifts. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    z_src_error = True
+
+            if (lens_gal_df['majorAxis'] - lens['REFF']/np.sqrt(1-lens['ELLIP'])) > 0.005:
+                if lens_major_axis_error is False:
+                    errors_string = errors_string + "\nLens Major Axis. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    lens_major_axis_error = True
+
+            if (lens_gal_df['minorAxis'] - lens['REFF']/np.sqrt(1+lens['ELLIP'])) > 0.005:
+                if lens_minor_axis_error is False:
+                    errors_string = errors_string + "\nLens Minor Axis. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    lens_minor_axis_error = True
+
+            if (lens_gal_df['sedFilepath'] != lens['lens_sed'][0]):
```

Comment:
Do you mind adding a test on `os.path.isfile(lens_gal_df['sedFilepath'])` to make sure that the SED file exists?  I know that I added a commit a few weeks ago that breaks the SED filenames output by the sprinkler.  I will fix it once we have a viable copy of cosmoDC2 to test on.  I would sleep easier if I knew SED file existence was being tested on (unless, I guess, it proves to be too expensive).

Reply:
These SEDs should just be in the `sims_SED_library`. Do you think that's necessary to test?

Reply:
Yes.  The damage I did in `Twinkles` revolved around where the SED subdir (`galaxySED`) gets prepended to the filename.  In DC1, the InstanceCatalog class added `galaxySED` through a call to `sims.utils.defaultSpecMap`.  In order to speed up SED matching in DC2, I moved that step into the SED fitter.

https://github.com/LSSTDESC/sims_GCRCatSimInterface/commit/74a4f32b2d3e0b7c38b5a558dfd3d749b17adf26